### PR TITLE
Use the project display name in FAR to support grouping

### PR DESCRIPTION
**Pull request by Pilchie**
Also, update ProjectDisplayName to do what the docs say it does for Venus files.

Partially addresses #18382.

### CODE REVIEW

**Comment by Pilchie** [comment]
Tagging @dotnet/roslyn-ide for review.

File: src/VisualStudio/Core/Def/Implementation/ProjectSystem/MiscellaneousFilesWorkspace.HostProject.cs [hunk 1]
```diff
@@ -77,7 +77,8 @@ public ProjectInfo CreateProjectInfoForCurrentState()
 
             public Workspace Workspace => _workspace;
 
-            public string ProjectSystemName => "MiscellaneousFiles";
+            public string DisplayName => "MiscellaneousFiles";
```
Comment (jasonmalinowski) [inline src/VisualStudio/Core/Def/Implementation/ProjectSystem/MiscellaneousFilesWorkspace.HostProject.cs]: Does this need to be localized or whatever?

File: src/VisualStudio/Core/Def/Implementation/ProjectSystem/AbstractProject.cs [hunk 2]
```diff
@@ -122,7 +122,10 @@ internal abstract partial class AbstractProject : ForegroundThreadAffinitizedObj
             ContentTypeRegistryService = componentModel.GetService<IContentTypeRegistryService>();
 
             this.RunningDocumentTable = (IVsRunningDocumentTable4)serviceProvider.GetService(typeof(SVsRunningDocumentTable));
-            this.DisplayName = projectSystemName;
+
+            var displayName = hierarchy != null && hierarchy.TryGetName(out var name) ? name : projectSystemName;
+            this.DisplayName = displayName;
```
Comment (jasonmalinowski) [inline src/VisualStudio/Core/Def/Implementation/ProjectSystem/AbstractProject.cs]: Why do we set this if it's also set in `UpdateProjectDisplayNameAndPath`?

**Comment by Pilchie** [comment]
Tagging @CyrusNajmabadi to see if he wants to respond to the feedback - he wrote this commit at my computer :)

**Comment by jasonmalinowski** [comment]
@Pilchie That explains the different author/committer. :smile:

**Comment by Pilchie** [comment]
Yeah, I ran commit with `--author=cyrusn@microsoft.com`

**Pull request merged**